llvm/lib/Target/AArch64/AArch64InstrInfo.cpp

```diff
@@ -6924,14 +6924,22 @@ genSubAdd2SubSub(MachineFunction &MF, MachineRegisterInfo &MRI,
     assert((Opcode == AArch64::SUBWrr || Opcode == AArch64::SUBXrr) &&
            "Unexpected instruction opcode.");
 
+  uint32_t RootFlags = Root.getFlags();
```

Comment:
Perhaps use `Root.mergeFlagsWith(*AddMI);`, clear the NoSWrap/NoUWrap on it they use that in the new instructions?